src/main/java/com/iota/iri/storage/Persistable.java

```diff
@@ -2,10 +2,49 @@
 
 import java.io.Serializable;
 
+/**
+ * Any object that can be serialized and stored in the DB.
+ *
+ * @see Indexable
+ */
 public interface Persistable extends Serializable {
+    /**
+     * Serializes the core fields of the object to bytes that can be stored in a database. Core fields are all the
+     * fields that represent the object in other nodes in the network.
+     *
+     * @return the serialized bytes that represents the core original object
+     */
     byte[] bytes();
+
+    /**
+     * Recreates the core state (populates the fields) of the object by deserializing {@code bytes}.
+     *
+     * @param bytes the serialized version the object
+     */
     void read(byte[] bytes);
+
+    /**
+     * Serializes fields that usually contain data about the object that the node itself calculates
+     * and is not passed to different nodes. This data is supposed to be stored in a different column family/table then
```

Comment:
```suggestion
     * and is not passed to different nodes. This data is stored in a different column family/table then
```
Its not supposed to right? We always do this.